File: Completed/JavaScript/807.js
/**
 * @param {number[][]} grid
 * @return {number}
 */
var maxIncreaseKeepingSkyline = function(grid) {
  let tb = new Array(grid.length).fill(0)
  let lr = new Array(grid.length).fill(0)
  
  for(let a = 0; a < grid.length;a++) {
      for(let b = 0; b < grid[a].length;b++) {
          tb[a] = Math.max(tb[a], grid[a][b])
          lr[b] = Math.max(lr[b], grid[a][b])
      }
  }
  let res = 0
  for(let a = 0; a < grid.length;a++) {
      for(let b = 0; b < grid[a].length;b++) {
          let diff = Math.min(tb[a], lr[b])
          res += diff - grid[a][b]
      }
  }
  return res
};

File: Completed/Go/807.go
func maxIncreaseKeepingSkyline(grid [][]int) int {
	rowMax := make([]int, len(grid))
	colMax := make([]int, len(grid[0]))

	for i := 0; i < len(grid); i++ {
		for j := 0; j < len(grid[i]); j++ {
			rowMax[i] = max(grid[i][j], rowMax[i])
			colMax[j] = max(grid[i][j], colMax[j])
		}
	}

	result := 0
	for i := 0; i < len(grid); i++ {
		for j := 0; j < len(grid[i]); j++ {
			m := min(rowMax[i], colMax[j])
			g := grid[i][j]

			result += m - g
		}
	}

	return result
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

func min(a, b int) int {
	if a < b {
		return a
	}
	return b
}